File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        List<int> digits = new List<int>();
        int origin = number;

        while (number != 0)
        {
            digits.Add(number % 10);
            number = (int)Math.Floor(number / (double)10);
        }

        int sum = digits.Sum(x => (int)Math.Pow(x, digits.Count));
        
        return sum == origin;
    }
}

File: cpp/armstrong-numbers/armstrong_numbers.cpp
#include "armstrong_numbers.h"
#include <math.h>

namespace armstrong_numbers
{

    bool is_armstrong_number(unsigned int n)
    {
        int digits = 0;
        unsigned int temp = n;

        do
        {
            ++digits;
            temp /= 10;
        } while (temp > 0);

        unsigned int sum = 0;
        temp = n;

        do
        {
            int digit = temp % 10;
            sum += static_cast<unsigned int>(std::pow(digit, digits));
            temp /= 10;
        } while (temp > 0);

        return sum == n;
    }

} // namespace armstrong_numbers
